File: lib/rate_limiter/rate_limiter_activity.ex
defmodule RateLimiter.Activity do
  use TemporalSdk.Activity
  @moduledoc false

  @impl true
  def execute(_context, [activity_count]) when is_integer(activity_count) do
    IO.puts("Executing activity number #{activity_count}")
    Process.sleep(1_000)
    ["ok"]
  end
end

File: src/rate_limiter/rate_limiter_activity.erl
-module(rate_limiter_activity).

% elp:ignore W0012 W0040
-moduledoc false.

-export([execute/2]).

-include_lib("temporal_sdk/include/activity.hrl").

execute(_Context, [ActivityCount]) when is_integer(ActivityCount) ->
    io:fwrite("Executing activity number ~b~n", [ActivityCount]),
    timer:sleep(1000),
    ["ok"].
